qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RoadObjectsActivity.kt

```diff
@@ -129,15 +129,15 @@ class RoadObjectsActivity : AppCompatActivity() {
         super.onStop()
         mapboxNavigation.unregisterRoutesObserver(routesObserver)
         mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
-        mapboxNavigation.unregisterRouteProgressObserver(replayProgressObserver)
         mapboxNavigation.unregisterLocationObserver(locationObserver)
     }
 
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
     override fun onDestroy() {
         super.onDestroy()
+        mapboxTripStarter.onDetached(mapboxNavigation)
```

Comment:
Is this necessary? Won't `onDetached` be called automatically when `MapboxNavigation` is destroyed?

Reply:
This example has a manual `MapboxNavigation` constructor so it needs manual attach and detach. The automatic `onDetached` only works with `MapboxNavigationApp`